### PR TITLE
benchalerts: add docstrings to mocked classes

### PR DESCRIPTION
**Pull request by austin3dickey**
This PR is in response to @jgehrcke's feedback on my last PR: https://github.com/conbench/conbench/pull/1362#discussion_r1266469361

This mocking strategy is probably not the best way to accomplish our goals. I'm sure there are other great alternatives I haven't looked into. This PR is only trying to document our current approach.

### CODE REVIEW

File: benchalerts/tests/unit_tests/mocks.py [hunk 1]
```diff
@@ -85,11 +105,20 @@ def send(self, *args, **kwargs):
...
     def __init__(self):
         super().__init__()
         self.session.mount("https://", MockAdapter())
 
     def _login_or_raise(self) -> None:
+        # Since we mock all requests, logging in won't change any state, so let's skip
+        # logging in. Logging in should be tested in the benchclients library already.
+        #
+        # (Also, as of this comment, the parent method sets up a new requests session,
+        # which would mean we'd lose our MockAdapter.)
```
Comment (alistaire47) [inline benchalerts/tests/unit_tests/mocks.py:121]: Just curious: Is there a reason this is a comment and not a docstring? Are we opposed to docstrings for internal methods?

**Review by alistaire47 — APPROVED** [review]

File: benchalerts/tests/unit_tests/mocks.py (hunk 1 above)
Comment (austin3dickey) [reply benchalerts/tests/unit_tests/mocks.py:121]: I was thinking this was more of a code comment about the next line, `pass`, than a description of this method. But either way would work.

File: benchalerts/tests/unit_tests/mocks.py [hunk 2]
```diff
@@ -41,6 +45,22 @@ def from_file(cls, file: pathlib.Path):
...
+    part of the filename is the HTTP method, the second is a key referring to a certain
+    base URL, and the rest is the path of the request URL, with special characters
+    replaced by underscores. Examples:
+
+    - GET https://conbench.biz/api/runs/?sha=abc -> GET_conbench_runs_sha_abc.json
+    - POST https://api.github.com/repos/some/repo/issues/1/comments ->
+        POST_github_issues_1_comments.json
+
+    This is intended to intercept ALL requests, and will raise an exception if the
+    request does not match any JSON file following the above format.
```
Comment (jgehrcke) [inline benchalerts/tests/unit_tests/mocks.py:61]: thank you!

**Review by jgehrcke — APPROVED** [review]
:rocket:

**Pull request merged**